adhocracy4/actions/signals.py

```diff
@@ -0,0 +1,53 @@
+from django.apps import apps
+from django.conf import settings
+from django.db.models.signals import post_save
+
+from adhocracy4.projects.models import Project
+
+from . import verbs
+from .models import Action
+
+
+def _has_project(instance):
+    return hasattr(instance, 'project') \
+           and instance.project.__class__ is Project
+
+
+def _extract_target(instance):
+    target = None
+    if hasattr(instance, 'content_object'):
+        target = instance.content_object
+    elif _has_project(instance):
```

Comment:
Following many other places (like our rules for instance) we could check here only for `hasattr(instance, 'project')`